synapse/config/workers.py

```diff
@@ -109,10 +109,12 @@ def read_config(self, config, **kwargs):
             federation_sender_instances
         )
 
-        # A map from instance name to host/port of their HTTP replication endpoint.
+        # A map from instance name to host/port of their HTTP replication endpoint
+        # (or None if there's no HTTP replication endpoint).
         instance_map = config.get("instance_map") or {}
         self.instance_map = {
-            name: InstanceLocationConfig(**c) for name, c in instance_map.items()
+            name: InstanceLocationConfig(**c) if c else None
+            for name, c in instance_map.items()
         }
```

Comment:
This might be abusing `instance_map` for what it was designed for? My hope was to somehow assert that `run_background_tasks_on` is a "real" worker. But maybe I should just drop this bit?

Reply:
I'd vote we drop this. There's nothing that verifies that things in the `instance_map` is running anyway, so I don't think it really helps at all.